packages/office-ui-fabric-react/src/components/Rating/Rating.tsx

```diff
@@ -90,12 +90,20 @@ export class Rating extends BaseComponent<IRatingProps, IRatingState> {
     this.setState({
       focusedRating: value
     } as IRatingState);
+
+    if (this.props.onFocus) {
```

Comment:
When _onFocus() is called, the element should have already obtained the focus.  At that moment, it needs to relay this event to user defined event handler.  The timing between relaying the event and state updated doesn't need strict order.    From testing on the example page, user's onFocus and onBlur event handler behaves as expected and no side effects were observed.